tests/test_physics/test_fock/test_fock.py

```diff
@@ -173,3 +173,14 @@ def test_dm_to_ket_error():
 
     with pytest.raises(ValueError):
         dm_to_ket(state)
+
+
+def test_fock_trace():
+    state = Vacuum(2) >> Ggate(2)
+    from_gaussian = state.get_modes(0).dm([3])
+    from_fock = State(dm=state.dm([40])).get_modes(0).dm([3])
+    assert np.allclose(from_gaussian, from_fock, atol=1e-5)
+
+    from_gaussian = state.get_modes(1).dm([3])
+    from_fock = State(dm=state.dm([40])).get_modes(1).dm([3])
+    assert np.allclose(from_gaussian, from_fock, atol=1e-5)
```

Comment:
Here a test directly using the trace function should be added as well. This test should contain a known input and compare the expected output, in that way it directly checks the function and this current test is then checkin its integration with the rest of the API.

Reply:
ok I'll improve this test